crates/bevy_input/src/gamepad.rs

```diff
@@ -473,114 +485,386 @@ impl GamepadSettings {
     }
 }
 
-/// Settings for a [`GamepadButton`].
-///
-/// ## Usage
-///
-/// It is used inside of the [`GamepadSettings`] to define the threshold for a gamepad button
-/// to be considered pressed or released. A button is considered pressed if the `press`
-/// value is surpassed and released if the `release` value is undercut.
+/// Manages settings for gamepad buttons
 ///
-/// ## Updating
+/// + `press_threshold` is the button input value above which the button is considered pressed.
+/// + `release_threshold` is the button input value below which the button is considered released.
 ///
-/// The current value of a button is received through the [`GamepadEvent`]s or [`GamepadEventRaw`]s.
+/// Allowed values: `0.0 <= ``release_threshold`` <= ``press_threshold`` <= 1.0`
 #[derive(Debug, Clone)]
 pub struct ButtonSettings {
-    /// The threshold for the button to be considered as pressed.
-    pub press: f32,
-    /// The threshold for the button to be considered as released.
-    pub release: f32,
+    press_threshold: f32,
+    release_threshold: f32,
 }
 
 impl Default for ButtonSettings {
     fn default() -> Self {
         ButtonSettings {
-            press: 0.75,
-            release: 0.65,
+            press_threshold: 0.75,
+            release_threshold: 0.65,
         }
     }
 }
 
 impl ButtonSettings {
+    /// Creates a new `ButtonSettings` instance
+    ///
+    /// # Arguments
+    ///
+    /// + `press_threshold` - the value above which the button is considered pressed.
+    /// + `release_threshold` - the value below which the button is considered released.
+    ///
+    /// Restrictions:
+    /// + `0.0 <= ``release_threshold`` <= ``press_threshold`` <= 1.0`
+    ///
+    /// # Errors
+    ///
+    /// If the restrictions are not met, `InvalidButtonSetting` will be returned.
+    pub fn new(press_threshold: f32, release_threshold: f32) -> Result<Self> {
```

Comment:
The return type here looks wrong to me?

Reply:
Correct; should be (and now is) `Result<ButtonSettings, GamepadSettingsError>`.